src/storage/restrict.rs

```diff
@@ -177,6 +186,38 @@ where
     }
 }
 
+
+impl<'st, T, D> Storage<'st, T, D>
+where
+    T: Component,
+    D: Deref<Target = MaskedStorage<T>>,
+{
+    /// Builds an immutable `RestrictedStorage` out of a `Storage`. Allows deferred
+    /// unchecked access to the entity's component.
+    pub fn restrict<'rf>(
+        &'rf self,
+    ) -> RestrictedStorage<'rf, 'st, &BitSet, T, &T::Storage, NormalRestriction> {
+        RestrictedStorage {
+            bitset: &self.data.mask,
+            data: &self.data.inner,
+            entities: &self.entities,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Builds an immutable, parallel `RestrictedStorage`.
+    pub fn par_restrict<'rf>(
+        &'rf self,
+    ) -> RestrictedStorage<'rf, 'st, &BitSet, T, &T::Storage, ParallelRestriction> {
```

Comment:
Do we really need two methods/variants here?

Reply:
Now that I think about it, probably not.